GameOfLife.java

```diff
@@ -60,19 +69,39 @@ public static void play(String fileName) {
 	// This function assumes that the input file contains valid data, and does no input testing.
 	public static int[][] read(String fileName) {
 		In in = new In(fileName); // Constructs an In object for reading the input file
+		int count = 0;
 		int rows = Integer.parseInt(in.readLine());
 		int cols = Integer.parseInt(in.readLine());
 		int[][] board = new int[rows + 2][cols + 2];
-		//// Replace the following statement with your code.
-		return null;
+		while (!in.isEmpty()){
+			count++;
+			String fill = in.readLine();
+			if (fill.length() > 0){
+				for(int i = 0; i < fill.length(); i++){
+					if(fill.charAt(i) == 'x'){
+						board[count][i + 1] = 1;
+					}
+					else if(fill.charAt(i) == '.'){
+						board[count][i + 1] = 0;
+					}
+				}
+			}	
```

Comment:
No need this `if`, when `fill.length() <= 0` it will not iterate through the `for` loop at all.
```suggestion
				for(int i = 0; i < fill.length(); i++){
					if(fill.charAt(i) == 'x'){
						board[count][i + 1] = 1;
					}
					else if(fill.charAt(i) == '.'){
						board[count][i + 1] = 0;
					}
				}
```